doc/tutorials/gpu-passthru.rst

```diff
@@ -0,0 +1,147 @@
+
+This tutorial describes how to enable GVT-d in ACRN.
+
+Introduction
+*****************
+Intel GVT-d is one flavor of graphics virtualization approaches also
+names as Intel-Graphics-Device pass-through feature.
+It based on Intel VT-d technology,
+and some special graphics related configuration.
+This flavor allows direct assignment of an entire GPU’s prowess to a single user,
+passing the native driver capabilities through the hypervisor without any limitations.
+
+Hardware Platform Supported
+***************************
+Currently, ACRN has enabled GVT-d on the following platforms:
+
+* Kaby Lake
+* Whiskey Lake
+* Elkhart Lake
+
+BIOS Setting
+**********************
+
+Kaby Lake Platform
+==================
+* set "IGD Minimum Memory" to "64MB" in "Devices → Video → IGD Minimum Memory".
+
+Whiskey Lake Platform
+=====================
+* set "PM Support"  to "Enabled" in "chipset → System Agent (SA) Configuration → Graphics Configuration → PM support".
+* set "DVMT Pre-Allocated" to "64MB" in "chipset → System Agent (SA) Configuration → Graphics Configuration → DVMT Pre-Allocated"
+
+Elkhart Lake Platform
+=====================
+* set "DMVT Pre-Allocated" to "64MB" in "Intel Advanced Menu → System Agent(SA) Configuration → Graphics Configuration → DMVT Pre-Allocated"
+
+Pass-through GPU to Guest
+**************************
+
+1. Copy ``/usr/share/acrn/samples/nuc/launch_win.sh`` to ``install_win.sh``
+
+::
+
+   cp /usr/share/acrn/samples/nuc/launch_win.sh ~/install_win.sh
+
+2. Modify ``install_win.sh`` script to specify the Windows image you use.
+
+3. Modify ``install_win.sh`` script to enable GVT-d.
+
+Add the following commands before ``acrn-dm -A -m $mem_size -s 0:0,hostbridge \``
+
+::
+
+      gpudevice=`cat /sys/bus/pci/devices/0000:00:02.0/device`
+
+      echo "8086 $gpudevice" > /sys/bus/pci/drivers/pci-stub/new_id
+      echo "0000:00:02.0" > /sys/bus/pci/devices/0000:00:02.0/driver/unbind
+      echo "0000:00:02.0" > /sys/bus/pci/drivers/pci-stub/bind
+
+Replace ``-s 2,pci-gvt -G "$2" \`` with ``-s 2,passthru,0/2/0,gpu \``
+
+4. Run the ``launch_win.sh``.
+
+      .. note:: If you want to pass-through GPU to Clear Linux User VM, the steps are
+         the same as above except your script.
+
+Enable GVT-d GOP driver
+***********************
+
+Issue
+======
+When enabling GVT-d, guest OS couldn’t light up physical screen
+before OS driver load. So guest BIOS and
+grub GUI couldn’t be showed on the physical screen.
+
+Reason
+==========
+Physical display is initialized by GOP driver or VBIOS
+before OS driver load, but guest BIOS doesn’t have them.
+
+Solution
+==========
+Integrate GOP driver binary into OVMF as a Dxe driver,
+then guest OVMF could see GOP driver and run it in
+graphic pass-through environment.
+So physical display could be initialized
+by GOP and used by guest BIOS and guest grub.
+
+Steps
+======
+1. fetch ACRN OVMF
+
+::
+
+      git clone https://github.com/projectacrn/acrn-edk2.git
```

Comment:
Can we reference a tag that's known to work? I see there are a number of tags called `ovmf-acrn-vX.X`, will there be an `acrn-ovmf-v1.6.1` that we can use? I think we should ;-)

Reply:
`acrn-ovmf` has no tag. So `git clone https://github.com/projectacrn/acrn-edk2.git` is enough.

Reply:
I have PR to `acrn-ovmf`. It hasn't merge now. We need to wait for this PR to be merged.

Reply:
After my PR has merged. It's enough to use acrn-edk2 master branch.

Reply:
I assume master is a moving target, can we also add a tag/commit ID once it's been merged. If people also want to try with master, they can but at least they would have a known good version to use if they have any doubt.

Reply:
This suggestion is great. I will add the commit ID later.

Reply:
> I assume master is a moving target, can we also add a tag/commit ID once it's been merged. If people also want to try with master, they can but at least they would have a known good version to use if they have any doubt.

I don't think we need to add commit ID. It's our duty to ensure master branch can work.

Reply:
> I don't think we need to add commit ID. It's our duty to ensure master branch can work.

I think we do. People are free to try out the latest as well but it's much preferable to have a version that's known to work for sure.

Reply:
> > I don't think we need to add commit ID. It's our duty to ensure master branch can work.
> 
> I think we do. People are free to try out the latest as well but it's much preferable to have a version that's known to work for sure.

If later patches have new features, the customer will lose it. 
Do we need to give the commit ID of acrn-hypervidor, acrn-kernel?
It's our duty to ensure mainline branch can work. If mainline can't work, we need to fix the bug.

Reply:
@liujunming , We need have verified version of acrn-hypervisor and kernel as the base of this document, please help to specify the detailed version information in document.